src/core/containers/runtime/uv_provider.py

```diff
@@ -0,0 +1,183 @@
+    extra_env: Optional[Dict[str, str]] = None
+    context_timeout_s: float = 60.0
+
+    _process: subprocess.Popen | None = field(init=False, default=None)
+    _base_url: str | None = field(init=False, default=None)
+
+    def start_container(
+        self,
+        image: str,
+        port: Optional[int] = None,
+        env_vars: Optional[Dict[str, str]] = None,
+        **_: Dict[str, str],
+    ) -> str:
+        if self._process is not None and self._process.poll() is None:
+            raise RuntimeError("UVProvider is already running")
+
+        self.repo_id = image or self.repo_id
+
+        bind_port = port or self.port or self._find_free_port()
+
+        command = _create_uv_command(
+            self.repo_id,
+            self.host,
+            bind_port,
+            self.reload,
+            project_url=self.project_url,
+        )
+
+        env = os.environ.copy()
+        if self.extra_env:
+            env.update(self.extra_env)
+        if env_vars:
+            env.update(env_vars)
+
+        try:
+            self._process = subprocess.Popen(command, env=env)
+        except FileNotFoundError as exc:
+            raise RuntimeError(
+                "`uv` executable not found. Install uv from "
+                "https://github.com/astral-sh/uv and ensure it is on PATH."
+            ) from exc
+        except OSError as exc:
+            raise RuntimeError(f"Failed to launch `uv run`: {exc}") from exc
+
+        client_host = self.connect_host or (
+            "127.0.0.1" if self.host in {"0.0.0.0", "::"} else self.host
+        )
+        self._base_url = f"http://{client_host}:{bind_port}"
+        self.port = bind_port
+        return self._base_url
+
+    def wait_for_ready(self, base_url: str, timeout_s: float = 60.0) -> None:
+        if self._process and self._process.poll() is not None:
+            code = self._process.returncode
+            raise RuntimeError(
+                f"uv process exited prematurely with code {code}"
+            )
+
+        _poll_health(f"{base_url}/health", timeout_s)
+
+    def stop_container(self) -> None:
+        if self._process is None:
+            return
+
+        if self._process.poll() is None:
+            self._process.terminate()
+            try:
+                self._process.wait(timeout=10.0)
+            except subprocess.TimeoutExpired:
+                self._process.kill()
+                self._process.wait(timeout=5.0)
+
+        self._process = None
+        self._base_url = None
+
+    def start(self) -> str:
+        return self.start_container(self.repo_id, port=self.port)
+
+    def stop(self) -> None:
+        self.stop_container()
+
+    def wait_for_ready_default(self, timeout_s: float | None = None) -> None:
+        if self._base_url is None:
+            raise RuntimeError("UVProvider has not been started")
+        self.wait_for_ready(
+            self._base_url,
+            timeout_s or self.context_timeout_s,
```

Comment:
I would let only 1 way to define the timeout (either in `wait_for_ready` or `__init__` but not in both)